### ISSUE TITLE
Bigtable: Instance object cannot create a Table with app profile

### ISSUE THREAD

**Issue by mbrukman** [GoogleCloudPlatform/google-cloud-python#5597]
The [`Instance` object][instance] has a method [`table()`][instance#table] which only takes a table id, but not an app profile. This means it's not possible to create a `Table` object given an instance to be used with app profiles, even though the `Table` object constructor does take an optional app profile parameter.

Thus, instead of a simple quickstart as follows:

```python
from google.cloud import bigtable

# Warning: this code doesn't work (yet)
client = bigtable.Client(project=project_id)
instance = client.instance(instance_id)
table = instance.table(table_id, app_profile_id)    # clean, simple; alas, does not work
```

we have to do something slightly round-about:

```python
from google.cloud import bigtable

client = bigtable.Client(project=project_id)
instance = client.instance(instance_id)
table = bigtable.table.Table(table_id, instance, app_profile_id)   # awkward
```

Proposal: extend `Instance#table()` to take an optional `app_profile_id` parameter that would be forwarded to the `Table` constructor and hence allow the simpler code pattern above.

Thoughts?

/cc: @sduskis, @billyjacobson, @jabubake, @ghaisa

[instance]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/a5b05d7fc50a0107a0d4f1371f6a1ea4959370f4/bigtable/google/cloud/bigtable/instance.py#L37
[instance#table]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/a5b05d7fc50a0107a0d4f1371f6a1ea4959370f4/bigtable/google/cloud/bigtable/instance.py#L256

**Comment by sduskis** [comment]
Basically, `instasnce.table()` needs an optional app_profile_id parameter.

**Comment by tseaver** [comment]
FTR, closed via #5605.
